web/src/stream_data.ts

```diff
@@ -1093,7 +1093,25 @@ export function can_use_empty_topic(stream_id: number | undefined): boolean {
         topics_policy = realm.realm_topics_policy;
     }
     return (
-        topics_policy === settings_config.get_realm_topics_policy_values().allow_empty_topic.code
+        topics_policy ===
+            settings_config.get_stream_topics_policy_values().allow_empty_topic.code ||
+        topics_policy === settings_config.get_stream_topics_policy_values().empty_topic_only.code
+    );
+}
+
+export function can_use_empty_topic_only(stream_id: number | undefined): boolean {
+    if (stream_id === undefined) {
+        return false;
+    }
+    const sub = sub_store.get(stream_id);
+    assert(sub !== undefined);
+
+    let topics_policy = sub.topics_policy;
+    if (sub.topics_policy === settings_config.get_stream_topics_policy_values().inherit.code) {
+        topics_policy = realm.realm_topics_policy;
+    }
+    return (
+        topics_policy === settings_config.get_stream_topics_policy_values().empty_topic_only.code
     );
 }
```

Comment:
stream_data.is_empty_topic_only_channel(stream_id) would be a much clearer name for this.

(I think that name is clear whether the channel is explicitly such or via inheritance).